src/linear_algebra/quadratic_form/prod.lean

```diff
@@ -0,0 +1,114 @@
+/-
+Copyright (c) 2021 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import linear_algebra.quadratic_form.basic
+
+/-! # Quadratic form on product and pi types
+
+## Main definitions
+
+* `quadratic_form.prod Q₁ Q₂`: the quadratic form constructed elementwise on a product
+* `quadratic_form.pi Q`: the quadratic form constructed elementwise on a pi type
+
+## Main results
+
+* `quadratic_form.equivalent.prod`, `quadratic_form.equivalent.pi`: quadratic forms are equivalent
+  if their components are equivalent
+* `quadratic_form.pos_def.prod`, `quadratic_form.pos_def.prod`: quadratic forms are positive-
+  definite if their components are positive definite.
```

Comment:
Conversely, if a product is anisotropic, then all its factors are anisotropic. Would you mind including that?

Reply:
Added. I'll come back and add the missing `pi` versions of these later tonight.